### ISSUE TITLE
Barman not deleting old backups

### ISSUE THREAD

**Issue by deric** [EnterpriseDB/barman#274]
I've following `retention_policy`:
```
retention_policy = RECOVERY WINDOW OF 3 DAYS
retention_policy_mode = auto
```
however barman is keeping 4 consistent backups and doesn't seem to apply the policy. Is this expected behavior?

```
# barman list-backup psql01a-ca
psql01a-ca 20200408T040004 - Wed Apr  8 13:14:11 2020 - Size: 1.1 TiB - WAL Size: 19.0 GiB
psql01a-ca 20200407T040006 - Tue Apr  7 11:46:34 2020 - Size: 1.1 TiB - WAL Size: 123.6 GiB
psql01a-ca 20200406T040003 - Mon Apr  6 12:03:13 2020 - Size: 1.1 TiB - WAL Size: 116.4 GiB
psql01a-ca 20200405T040003 - Sun Apr  5 15:10:55 2020 - Size: 1.1 TiB - WAL Size: 94.5 GiB
```

**Comment by gbartolini** [comment]
When applying recovery window policies, Barman thinks in terms of point of recoverability.
In your case, based on recovery window, you are asking that Barman allows you to recover at any point in time between now and 3 days ago.
The point of recoverability at any time is "now - 3 days". This means that Barman needs to have the first valid backup prior to the point of recoverability.
It might be plausible that, in order to recover to "now - 3 days" ago, Barman requires the backup of 4 days ago and then roll forward redo logs.
There is not direct relationship with redundancy (number of backups).
Please let me know if this can be closed.

**Comment by deric** [comment]
How often is this evaluated? Can I trigger a manual clean-up? Ideally with a dry-run? I understand that for `now - 3 days` ago recoverability barman needs WAL files to replay DB transactions. But does it really need an extra full base backup?

Also, when I look at the `wal` directory, there are still WAL files there are still files more than 5 months old. Given the traffic on the server these can be for sure safely removed:
```
drwxr-xr-x 2 barman barman   16384 Apr  6 06:01 0000000100007892
drwxr-xr-x 2 barman barman   20480 Apr  6 05:21 0000000100007891
drwxr-xr-x 2 barman barman   16384 Apr  6 04:47 0000000100007890
drwxr-xr-x 2 barman barman   20480 Jan 16 11:50 0000000100006321
drwxr-xr-x 2 barman barman   20480 Dec  2 10:30 0000000200005D5E
drwxr-xr-x 2 barman barman   12288 Nov 22 13:27 0000000200005D74
drwxr-xr-x 2 barman barman   16384 Nov 22 12:58 0000000200005D73
drwxr-xr-x 2 barman barman   16384 Nov 22 12:24 0000000200005D72
drwxr-xr-x 2 barman barman   20480 Nov 22 12:08 0000000200005D71
drwxr-xr-x 2 barman barman   16384 Nov 22 11:43 0000000200005D70
drwxr-xr-x 2 barman barman   16384 Nov 22 11:23 0000000200005D6F
drwxr-xr-x 2 barman barman   20480 Nov 22 11:02 0000000200005D6E
```
I understand that `Apr 9 - 3 days = Apr 6` are needed. But WAL files from November last year?

**Comment by amenonsen** [comment]
You are, of course, correct that WAL from last November would not be required to restore your backups from April of this year. I suspect that these WAL files were not recorded in xlog.db because of the bug described in #328 (which was recently fixed in the Barman 2.13 release).

It's been quite a while since your last response, so this is probably no longer relevant, but I suspect that if you ran `barman rebuild-xlogdb` on your server, it would (a) record these old WAL segments in xlog.db, and (b) remove them at the next opportunity (e.g., at the next `barman backup` run).

Apart from this, we expect to provide more options around enforcing retention policies in a future release, so I'm going to close this issue now.
